src/io/platform/f4xx/I2Cf4xx.cpp

```diff
@@ -0,0 +1,267 @@
+#include <EVT/io/platform/f4xx/I2Cf4xx.hpp>
+
+#include <EVT/io/pin.hpp>
+#include <EVT/io/platform/f4xx/GPIOf4xx.hpp>
+
+#include <EVT/utils/log.hpp>
+
+//namespace log = EVT::core::log;
+
+namespace EVT::core::IO {
+
+/**
+ * Get the I2C ID associated that is associated with the given pin
+ * combination. This information is pulled from the STM32F302x8
+ * documentation with easier documentation on MBed's STM32F302r8
+ * documentation.
+ *
+ * TODO: Make modifications for this to be runnable at compile time
+ * allowing for validation of the selected pins.
+ *
+ * @param sclPin The selected I2C clock pin
+ * @return The port ID associated with the selected pins
+ */
+static uint8_t getPortID(Pin sclPin) {
+#ifdef STM32F302x8
```

Comment:
You should remove the `#ifdef`s for the 3xx chips because they'll never be used from this file